src/biogeochem/CNVegNitrogenStateType.F90

```diff
@@ -1815,11 +1833,20 @@ subroutine Restart ( this,  bounds, ncid, flag, leafc_patch, &
                    end if
                 end if 
              else
-                this%leafn_patch(p)                           = leafc_patch(p)         / pftcon%leafcn(patch%itype(p))
-                this%leafn_storage_patch(p)                   = leafc_storage_patch(p) / pftcon%leafcn(patch%itype(p))
+                ! max in next line ensures leafcn_patch equals at least the
+                ! leafcn for bare ground (noveg) to avoid div by 0 when the
+                ! initial file doesn't include leafcn_patch. In that situation
+                ! it seems, however, like a better initial condition to set
+                ! this%leafcn_patch(p) = pftcon%leafcn(patch%itype(p))
+                ! ...but leafcn_patch does get updated before anything else
+                ! in clm_driver, so either option may give the same answer.
```

Comment:
This may be unnecessary because I eliminated leafcn_patch from restart files.

So this backwards compatibility note and line 1843 are unnecessary.